### DISCUSSION TITLE
Less privileges on container creation

### DISCUSSION THREAD

**Discussion by mewid** [middlewarehq/middleware#392] [Ideas]
Hi team!

I'm trying to use the middleware application on k8s as a deployment, but it seems to need a lot of setup during creation. Like unzupping files and creating folders and such. This becomes a problem for us since we need to give the container full write access, which we normally would not want to do.

Would it be possible to do much of this bootstraping earlier, in the image creation? Due to security restrictions we need to give write access folder by folder, which is becoming complex.

**Comment by mewid** [comment]
We have now tried different solutions to host the container in our cluster, but we have come to a halt. Tell me if I'm wrong, but it seems like the container have to run as root user (or at least with some elevated privileges). Even if we give it full write access to the file system (which we would prefer not to), it is unable to unzip the file.

To run the container as root user is not an alternative for us.

**Reply by jayantbh** [reply]
Hey @mewid, apologies that this kinda dropped off the radar.

@adnanhashmi09 maybe you'd be able to help here?

**Reply by mewid** [reply]
Any update on this? We are eager to be able to utilise this within our team :)

**Reply by mewid** [reply]
Pinging once more, in case this fell through the cracks! @adnanhashmi09 @jayantbh (sorry for pinging you, but not sure if you get notified otherwise)

**Reply by jayantbh** [reply]
Hey @mewid, apologies. :)
I've brought this up internally and we'll check it out. Might need more discussion with you to sort this out.

While the discussion thread can work just fine, you can also ping us on [our OSS Slack](https://mhq.link/oss-community) for quicker back-and-forth, or in case we need to get on a quick call. 🚀

**Reply by mewid** [reply]
Sure, I will join the slack channel!
